src/dialogs/vkbasalt.py

```diff
@@ -0,0 +1,306 @@
+                self.smaa.set_enable_expansion(False)
+            # Check if clut is used.
+            if VkBasaltSettings.lut_file_path is None:
+                self.clut.set_enable_expansion(False)
+                self.lut_file_path = False
+            else:
+                self.input_entry.set_text(VkBasaltSettings.lut_file_path)
+                self.lut_file_path = VkBasaltSettings.lut_file_path
+
+            if VkBasaltSettings.cas_sharpness != None:
+                self.cas_sharpness.set_value(float(VkBasaltSettings.cas_sharpness))
+            if VkBasaltSettings.dls_sharpness != None:
+                self.dls_sharpness.set_value(float(VkBasaltSettings.dls_sharpness))
+            if VkBasaltSettings.dls_denoise != None:
+                self.dls_denoise.set_value(float(VkBasaltSettings.dls_denoise))
+            if VkBasaltSettings.fxaa_subpixel_quality != None:
+                self.fxaa_subpixel_quality.set_value(float(VkBasaltSettings.fxaa_subpixel_quality))
+            if VkBasaltSettings.fxaa_quality_edge_threshold != None:
+                self.fxaa_quality_edge_threshold.set_value(float(VkBasaltSettings.fxaa_quality_edge_threshold))
+            if VkBasaltSettings.fxaa_quality_edge_threshold_min != None:
+                self.fxaa_quality_edge_threshold_min.set_value(float(VkBasaltSettings.fxaa_quality_edge_threshold_min))
+            if VkBasaltSettings.smaa_threshold != None:
+                self.smaa_threshold.set_value(float(VkBasaltSettings.smaa_threshold))
+            if VkBasaltSettings.smaa_max_search_steps != None:
+                self.smaa_max_search_steps.set_value(float(VkBasaltSettings.smaa_max_search_steps))
+            if VkBasaltSettings.smaa_max_search_steps_diagonal != None:
+                self.smaa_max_search_steps_diagonal.set_value(float(VkBasaltSettings.smaa_max_search_steps_diagonal))
+            if VkBasaltSettings.smaa_corner_rounding != None:
+                self.smaa_corner_rounding.set_value(float(VkBasaltSettings.smaa_corner_rounding))
+            if VkBasaltSettings.disable_on_launch == "True":
+                self.switch_disable_on_launch.set_state(True)
+            if VkBasaltSettings.smaa_edge_detection != None:
+                if VkBasaltSettings.smaa_edge_detection == "color":
+                    self.color.set_active(True)
+                    self.smaa_edge_detection = "color"
+                else:
+                    self.smaa_edge_detection = "luma"
+            else:
+                self.smaa_edge_detection = "luma"
+        else:
+            self.default.set_state(True)
+            self.smaa_edge_detection = "luma"
+            self.cas.set_enable_expansion(False)
+            self.dls.set_enable_expansion(False)
+            self.fxaa.set_enable_expansion(False)
+            self.smaa.set_enable_expansion(False)
+            self.clut.set_enable_expansion(False)
+            self.lut_file_path = False
+
+    def __idle_save(self, *args):
```

Comment:
The callback of a idle_add has to return a boolean, it can be either GLib.SOURCE_CONTINUE or GLib.SOURCE_REMOVE. If you only want to run this once, use the later.